onmt/inputters/inputter.py

```diff
@@ -209,306 +12,135 @@ class IterOnDevice(object):
         self.iterable = iterable
         self.device_id = device_id
+        self.transforms = iterable.transforms
 
     @staticmethod
-    def batch_to_device(batch, device_id):
+    def batch_to_device(tbatch, device_id):
         """Move `batch` to `device_id`, cpu if `device_id` < 0."""
-        curr_device = batch.indices.device
         device = torch.device(device_id) if device_id >= 0 \
             else torch.device('cpu')
-        if curr_device != device:
-            if isinstance(batch.src, tuple):
-                batch.src = tuple([_.to(device) for _ in batch.src])
-            else:
-                batch.src = batch.src.to(device)
-            batch.tgt = batch.tgt.to(device)
-            batch.indices = batch.indices.to(device)
-            batch.alignment = batch.alignment.to(device) \
-                if hasattr(batch, 'alignment') else None
-            batch.src_map = batch.src_map.to(device) \
-                if hasattr(batch, 'src_map') else None
-            batch.align = batch.align.to(device) \
-                if hasattr(batch, 'align') else None
+        for key in tbatch.keys():
+            if key != 'src_ex_vocab':
+                tbatch[key] = tbatch[key].to(device)
 
     def __iter__(self):
-        for batch in self.iterable:
-            self.batch_to_device(batch, self.device_id)
-            yield batch
-
-
-def filter_example(ex, use_src_len=True, use_tgt_len=True,
-                   min_src_len=1, max_src_len=float('inf'),
-                   min_tgt_len=1, max_tgt_len=float('inf')):
-    """Return whether an example is an acceptable length.
-
-    If used with a dataset as ``filter_pred``, use :func:`partial()`
-    for all keyword arguments.
-
-    Args:
-        ex (torchtext.data.Example): An object with a ``src`` and ``tgt``
-            property.
-        use_src_len (bool): Filter based on the length of ``ex.src``.
-        use_tgt_len (bool): Similar to above.
-        min_src_len (int): A non-negative minimally acceptable length
-            (examples of exactly this length will be included).
-        min_tgt_len (int): Similar to above.
-        max_src_len (int or float): A non-negative (possibly infinite)
-            maximally acceptable length (examples of exactly this length
-            will be included).
-        max_tgt_len (int or float): Similar to above.
-    """
-
-    src_len = len(ex.src[0])
-    tgt_len = len(ex.tgt[0])
-    return (not use_src_len or min_src_len <= src_len <= max_src_len) and \
-        (not use_tgt_len or min_tgt_len <= tgt_len <= max_tgt_len)
-
-
-def _pad_vocab_to_multiple(vocab, multiple):
-    vocab_size = len(vocab)
-    if vocab_size % multiple == 0:
-        return
-    target_size = int(math.ceil(vocab_size / multiple)) * multiple
-    padding_tokens = ["{}{}".format(DefaultTokens.VOCAB_PAD, i)
-                      for i in range(target_size - vocab_size)]
-    vocab.extend(Vocab(Counter(), specials=padding_tokens))
-    return vocab
```

Comment:
This might still be useful when using mixed precision.

Reply:
Yes I know I skipped this.

Reply:
done